torch/csrc/distributed/rpc/metrics/RpcMetricsHandler.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <string>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+// All metrics are prefixed with the following  key.
+const std::string kRPCMetricsKeyPrefix = "torch.distributed.rpc.";
+// Thread-safe API for logging time-series metrics for RPC-based model parallel
+// training. This class provides thread safety so that metrics can be logged
+// from multiple threads without the user needing to coordinate serialization.
+// The class provides two main functions - accumulateMetric which can be used
+// for statistical aggregations and setMetric for setting/modifying single
+// values.
```

Comment:
We should mention that implementations of this class should provide these APIs in a thread safe manner. Since this class is just an interface, it can't provide any Thread safety guarantees.